sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_publisher_client_async.py

```diff
@@ -73,7 +78,7 @@ class EventGridPublisherClient:
     def __init__(
         self,
         endpoint: str,
-        credential: Union[AzureKeyCredential, AzureSasCredential],
+        credential: Union["AsyncTokenCredential", AzureKeyCredential, AzureSasCredential],
```

Comment:
That seems inconsistent, if not used those types should be in the `TYPE_CHECKING` as well, but I see now reason why some type would be string and some would be types